src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/InductionEntityMapper.kt

```diff
@@ -61,11 +61,26 @@ abstract class InductionEntityMapper {
   @Mapping(target = "updatedAtPrison", source = "prisonId")
   abstract fun fromCreateDtoToEntity(dto: CreateInductionDto): InductionEntity
 
-  @Mapping(target = "lastUpdatedBy", source = "updatedBy")
-  @Mapping(target = "lastUpdatedByDisplayName", source = "updatedByDisplayName")
-  @Mapping(target = "lastUpdatedAt", source = "updatedAt")
-  @Mapping(target = "lastUpdatedAtPrison", source = "updatedAtPrison")
-  abstract fun fromEntityToDomain(persistedEntity: InductionEntity): Induction
+  fun fromEntityToDomain(inductionEntity: InductionEntity, previousQualificationsEntity: PreviousQualificationsEntity?): Induction =
```

Comment:
Here's an example of where MapStruct mappers don't need to be generated classes (or more specifically dont need to be entirely generated)
In respect of this change I started off by adding more `@Mapping` annotations to the method signature to reflect how the mapping needs to work. It ended up being an unnecessary amount of annotations for what it was worth, and was easier and neater to simply write the implementation as I have done here.

The point I guess I'm making is that we can treat these mapping functions on a case by case basis. Where there's value in writing the implementation ourselves then we can. Equally where it's easier/neater to just write the method sig and let MapStruct generate the implementation then we can do that as well.
It's flexible 👍

Reply:
Looks good, I agree with providing an impl. My tolerance for keeping mapstruct on some of these is probably lower than yours and if it was up to me, anything more than a couple of annotations or any `java foo(entity, dto)` type mappings should be purged.